plugin/trino-password-authenticators/src/main/java/io/trino/plugin/password/ldap/JdkLdapClient.java

```diff
@@ -106,11 +108,30 @@ public boolean isGroupMember(String searchBase, String groupSearch, String conte
         }
     }
 
+    @Override
+    public Set<String> lookupUserGroups(String searchBase, String searchFilter, String contextUserDistinguishedName, String contextPassword)
+            throws NamingException
+    {
+        try (CloseableContext context = createUserDirContext(contextUserDistinguishedName, contextPassword);
+                CloseableSearchResults search = searchContext(searchBase, searchFilter, context)) {
+            ImmutableSet.Builder<String> groupNames = ImmutableSet.builder();
+            if (search.hasMore()) {
+                Attributes attributes = search.next().getAttributes();
+                Attribute memberOfAttribute = attributes.get("memberof");
+                if (memberOfAttribute == null) {
+                    throw new NamingException("No memberOf attribute found... The ldap group provider requires the memberOf overlay to be enabled.");
+                }
```

Comment:
Where are we adding the group names here ?

Reply:
So we don't support groups from LDAP server doesn't provide the `memberof` ? In that case can we fetch the groups based on DN of the user ?